LocalPackages/AppInfoRetriever/Sources/AppInfoRetriever/AppInfoRetriever.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  AppInfoRetriever.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import AppKit
+import Foundation
+
+public protocol AppInfoRetrieveing {
+
+    /// Provides a structure featuring commonly-used app info.
+    ///
+    /// It's also possible to retrieve the individual information directly by calling other methods in this class.
+    ///
+    func getAppInfo(bundleID: String) -> AppInfo?
+    func getAppInfo(appURL: URL) -> AppInfo?
+    func getAppIcon(bundleID: String) -> NSImage?
+    func getAppName(bundleID: String) -> String?
+    func getBundleID(appURL: URL) -> String?
+
+}
+
+public class AppInfoRetriever: AppInfoRetrieveing {
```

Comment:
Neat – we could possibly reuse this for the Autofill feature, which does some app info lookup of its own but in a non-generic way.